### DISCUSSION TITLE
configuration question: Using Raspberry Pi as IP camera + Motionplus on server

### DISCUSSION THREAD

**Discussion by sroller** [Motion-Project/motionplus#187] [Q&A] [answered]
I'm trying to set up several PIs with PI camera modules all streaming to a x86 Ubuntu 24.04.x server.
I can stream using libcamera-vid + gs-launch-1.0 or ffmpeg via RTP to VLC.
I spent several hours finding a solution that works with Motionplus to no avail. It's poorly documented on PI side and on the motionplus side. There are only little bits here and there.
Is there anybody who has a working environment and would be willing to share their configuration?
Is RTP even the best solution to achieve this? Should I go with RTMS or mjpeg instead?

I have another suggestion: Could we have a wiki on this project? I know github has that option. Perhaps that could help to make the knowledge of the community in a more structured way. My apologies if there's already one and haven't found it yet.

-st

**Comment by axeljerabek** [comment, marked as answer]
Hi sroller,
I have several Pis of different makes with different cam modules streaming over nginx to my ubuntu 24 machine.
Maybe you find something here:
https://www.wetransco.de/2023/12/raspberry-pi-zero-2w-debian-bookworm-und-libcamera-module-2-3-streaming/

You can also contact me if you run into problems somewhere. This setup works since quite a while and is pretty stable.
Good luck!

**Reply by axeljerabek** [reply]
Sorry, that artifcle is also available in english somewhere :)
Here it is:
https://www.wetransco.de/2023/12/debian-bookworm-and-libcamera-module-3/

**Reply by sroller** [reply]
Deutsch ist meine Muttersprache :-D

**Comment by sroller** [comment]
thank you, I got it working. there were some inconsistencies in the description but that allowed me to learn :-).

**Reply by axeljerabek** [reply]
Sehr gut. War noch von einer älteren Version, oder welche Inkonsistenz meintest du? :)
Übrigens ein echter Vorteil in der Methode: Ich nutze den h264 encoder auf der Kamera selbst, so dass ich an keiner Stelle mehr ein Encoding oder Transcoding nutzen muss. Das spart enorm Rechenleistung, weshalb eben auch ein Rpi Zero einsetzbar ist.
